packages/stream/index.d.ts

```diff
@@ -40,11 +40,17 @@ export function prime<
   TReturn = unknown,
 >(
   writer: AsyncGenerator<TRead,  TReturn, TWrite>,
-  // Omitted from type definition because undefined is
-  // nearly always adequate:
-  // primer: TWrite,
+  primer: TWrite,
 ): Stream<TRead, TWrite, TReturn, TReturn>;
 
+// Overload:
+export function prime<
+  TRead,
+>(
+  writer: AsyncGenerator<TRead, undefined, undefined>,
+  // primer is implicitly undefined for this overload.
+): Stream<TRead, undefined, undefined, undefined>;
+
```

Comment:
TS chooses the overload using top down, and uses the first one that matches. In this case, I'm not sure but the first one may actually match, and TS may pick it with `TWrite` defaulting to `unknown`.

Reply:
Addressed.